Automattic/woocommerce-payments
# Move `usePaymentActivityData()` to `PaymentActivity` because filters will need it

Fixes https://github.com/Automattic/woocommerce-payments/issues/8733

#### Changes proposed in this Pull Request

* Move `usePaymentActivityData()` call to `<PaymentActivity>` from `<PaymentActivityData>`.

Because we are going to show filters in the Payment Activity widget, the payment activity data fetch (the call to `usePaymentActivityData()`) needs to happen in the same component of the filter, in `<PaymentActivity>`.
This is needed because while `isLoading` true, I like to either disable or hide the filters and it's hard to do if the `isLoading` is within the child element of `<PaymentActivity>`.

note: I realize that with this change, Payments > Overview page will try to fetch payment activity data even though lifetime TPV is 0 where with base branch, the data fetch will only happen only if lifetime TPV > 0.

* Rename component `<PaymentActivityData>` to `<PaymentActivityDataComponent>` so it does not conflict with the `PaymentActivityData` type.

* Use the date start and end from payment activity data returned by the backend to generate the report links.

#### Testing instructions

* Enable the feature flag `_wcpay_feature_payment_overview_widget`.
* Make sure Payments > Overview > Payment Activity widget shows up correctly.

##### Testing error handling when server does not return proper response.

* Edit `includes/admin/class-wc-rest-payments-reporting-controller.php` locally so it returns an error to frontend.

On this [line](https://github.com/Automattic/woocommerce-payments/blob/56f176df3fff5caa5b5df030f3fd93691ec1c5d2/includes/admin/class-wc-rest-payments-reporting-controller.php#L48):
```diff
-		$wcpay_request->set_date_start( $request->get_param( 'date_start' ) );
+		$wcpay_request->set_date_start( 'abc' );
```

* Make sure Payments > Overview > Payment Activity widget disappears after loading animation.

-------------------


**Post merge**
